public/app/plugins/datasource/loki/components/LokiContextUi.test.tsx

```diff
@@ -113,20 +113,20 @@ describe('LokiContextUi', () => {
   });
 
   it('calls updateFilter when selecting a label', async () => {
-    jest.useFakeTimers();
     const props = setupProps();
     render(<LokiContextUi {...props} />);
     await waitFor(() => {
       expect(props.languageProvider.start).toHaveBeenCalled();
     });
     const select = await screen.findAllByRole('combobox');
     await selectOptionInTest(select[1], 'label3');
-    act(() => {
-      jest.runAllTimers();
-    });
-    expect(props.updateFilter).toHaveBeenCalled();
-
-    jest.useRealTimers();
+    // TODO whyyyy do we have a 1500ms timeout in the component?!
```

Comment:
Probably cause of this code in the component: 
```
    timerHandle.current = window.setTimeout(() => {
      updateFilter(contextFilters);
      setLoading(false);
    }, 1500);
```
Using fake timers is a preferred approach here, otherwise we're basically adding almost 2sec to the test execution time. Can't figure out why using `fakeTImers` doesn't work in this case, but in general it seems like the tests are testing the implementation details and not the whole component behavior. I suggest skipping this test until we can fogure out a better alternative.

Reply:
fyi @svennergr

Reply:
yeah sorry i was more venting about not being able to get fake timers to work than anything 😂 i've skipped it for now with a comment. if i can get fake timers working again i'll use that instead 👍

Reply:
got it working with https://github.com/grafana/grafana/pull/64428/commits/b03b23a141aa7e1faf9156ac1d1f66759e0efc16

Reply:
No idea why that code works, but the previous version doesn't 😂

Reply:
it's cause `const select = await screen.findAllByRole('combobox');` was only finding 1 combobox, so the element passed to `selectOptionInTest` was `undefined` 

by adding `expect(screen.getAllByRole('combobox')).toHaveLength(2);` in the `waitFor` we ensure we wait for both comboboxes to be visible

Reply:
`findAllBy*` is a wrapper for `waitFor` + `getBy*`, so it's weird that I didn't work, but using `getBy` inside `waitFor` does.